core/src/main/java/com/scalar/db/api/ScanBuilder.java

```diff
@@ -936,525 +905,299 @@ public Scan build() {
+        BuildableScanFromExistingWithOngoingWhere buildable) {
       super(buildable);
     }
 
-    private BuildableScanAllFromExistingWithOngoingWhereAnd(
+    private BuildableScanFromExistingWithOngoingWhereAnd(
         BuildableScanOrScanAllFromExisting buildable, OrConditionSet orConditionSet) {
       super(buildable);
-      disjunctions.add(orConditionSet.getConditions());
+      where.and(orConditionSet);
     }
 
-    private BuildableScanAllFromExistingWithOngoingWhereAnd(
+    private BuildableScanFromExistingWithOngoingWhereAnd(
         BuildableScanOrScanAllFromExisting buildable, Set<OrConditionSet> orConditionSets) {
       super(buildable);
-      disjunctions.addAll(
-          orConditionSets.stream().map(OrConditionSet::getConditions).collect(Collectors.toSet()));
+      where.and(orConditionSets);
     }
 
     @Override
-    public BuildableScanAllFromExistingWithOngoingWhereAnd and(ConditionalExpression condition) {
+    public BuildableScanFromExistingWithOngoingWhereAnd and(ConditionalExpression condition) {
       checkNotNull(condition);
-      disjunctions.add(ImmutableSet.of(condition));
+      where.and(condition);
       return this;
     }
 
     @Override
-    public BuildableScanAllFromExistingWithOngoingWhereAnd and(OrConditionSet orConditionSet) {
+    public BuildableScanFromExistingWithOngoingWhereAnd and(OrConditionSet orConditionSet) {
       checkNotNull(orConditionSet);
-      disjunctions.add(orConditionSet.getConditions());
+      where.and(orConditionSet);
       return this;
     }
   }
 
-  /**
-   * An and-wise set of {@link ConditionalExpression} used for specifying arbitrary conditions in a
-   * {@link Scan} command.
-   */
-  @Immutable
-  public static class AndConditionSet {
-    private final ImmutableSet<ConditionalExpression> conditions;
-
-    private AndConditionSet(ImmutableSet<ConditionalExpression> conditions) {
-      this.conditions = conditions;
-    }
-
-    /**
-     * Returns the set of {@code ConditionalExpression}.
-     *
-     * @return set of {@code ConditionalExpression}
-     */
-    public Set<ConditionalExpression> getConditions() {
-      return conditions;
-    }
-
-    /**
-     * Indicates whether some other object is "equal to" this object. The other object is considered
-     * equal if:
-     *
-     * <ul>
-     *   <li>it is also an {@code AndConditionSet}
-     *   <li>both instances have the same set of {@code ConditionalExpression}
-     * </ul>
-     *
-     * @param o an object to be tested for equality
-     * @return {@code true} if the other object is "equal to" this object otherwise {@code false}
-     */
-    @Override
-    public boolean equals(Object o) {
-      if (o == this) {
-        return true;
-      }
-      if (!(o instanceof AndConditionSet)) {
-        return false;
-      }
-      AndConditionSet other = (AndConditionSet) o;
-      return conditions.equals(other.conditions);
-    }
+  private static class OngoingWhere {
```

Comment:
This is introduced for simply and commonly managing the internal condition state for both `BuildableScanAllWithWhere` and `BuildableScanFromExistingWithXX`. Since it can also be used for `GetBuilder,` this class may be moved to somewhere commonplace.

Reply:
Also, I avoided using just `Where` to distinguish the `Where` interface, but the name might be a bit weird since it will also be used after closing the where clause. `InternalWhere`, `WhereClause`, or `WhereBuilder`? What do you think?

Reply:
`WhereBuilder` sounds to me that it has `build()` method. So, it might be worth adding `build()` that returns `Collection<Conjunction>` or something.

`WhereClause` itself sounds good to me, but I noticed the term `Clause` isn't used in this file...

I think `WhereCondition` might be an option as the class contains only conditions.

Reply:
Thanks for the suggestion. I changed my mind a bit. Since the interface `Where` is only used twice in this file, using it with the modifier (i.e., `OperationBuilder.Where`) is not so bad looking. I'm going with this if no further concerns.